### PR TITLE
FIX-#4113, FIX-#4116, FIX-#4115: Apply new `black` formatting, fix pydocstyle check and readthedocs build

### LINKED ISSUE #4113: Update code formatting to align with new `black` version

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

These changes fix the failed CI. Issues list:

1. #4113 
2. #4115 
3. #4116 

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4113, #4115, #4116  
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by dchigarev** [comment]
Maybe we should pin the actual version of `black` and update it periodically, rather than instantly getting failing CI every time `black` releases a new version. @prutskov @modin-project/modin-core, what do you think?

**Comment by prutskov** [comment]
> Maybe we should pin the actual version of `black` and update it periodically, rather than instantly getting failing CI every time `black` releases a new version. @prutskov @modin-project/modin-core, what do you think?

I agree that it will give more control on new releases. Also, in the case we pin `black` (flake8?) it would be better to add its to our development dependencies to free the user from installing of `black` package with specific version. I can add this in this PR.
@dchigarev @modin-project/modin-core, thoughts?

**Comment by prutskov** [comment]
An issue related readthedocs failure #4115.

**Comment by YarShev** [comment]
Yes, I think that is good option to add concrete black/flake8 versions in our env files and upgrading them staged.

**Comment by YarShev** [comment]
Though that will require us constantly monitoring for new releases of black and flake8 if we want to have our code formatted in line with latest versions.

**Comment by prutskov** [comment]
> Though that will require us constantly monitoring for new releases of black and flake8 if we want to have our code formatted in line with latest versions.

I agree, but it will be not often action due to flake8/black are rarely released. From the one hand, the pinning adds monitoring activity and expands our env files. From the other hand, it excludes unexpected CI failure in the most important situation (before release, for example) and gives us more control on new versions.

I've added pins for black/flake8 in our env files for now. @YarShev, @dchigarev, @modin-project/modin-core, @modin-project/modin-omnisci, what do you think?

File: environment-dev.yml
Comment (dchigarev) [inline environment-dev.yml:45]: but the actual CI recipe [doesn't use any of these](https://github.com/prutskov/modin/blob/00af9ab95c5c7b2ca9859e86d48dc7d9b1fba981/.github/workflows/ci.yml#L37)

**Comment by dchigarev** [comment]
If we decided to pin a black version, we should explicitly voice that somewhere in the developer's docs/PR template. Currently, it requires running **any** `black` and `flake` against Modin repo before doing a PR, although we depend on their certain versions there.

File: environment-dev.yml
Comment (prutskov) [reply environment-dev.yml]: not relevant

**Comment by prutskov** [comment]
@dchigarev, @YarShev, I propose to do changes related pinning of flake/black in a separate issue to not delay original issue related to immediate fix of CI.

File: scripts/doc_checker.py
Comment (dchigarev) [inline scripts/doc_checker.py:49]: the `dask` import flow has suddenly [stopped working](https://github.com/dchigarev/modin/runs/5005582579?check_suite_focus=true#step:9:56) in the doc-checker script, we may also want to check whether it causes a problem in a regular `import modin.pandas as pd` import flow with the dask engine in a clean env, as a part of a separate issue ofc.

**Review by dchigarev — DISMISSED** [review]

File: scripts/doc_checker.py
Comment (dchigarev) [reply scripts/doc_checker.py:49]: I think we can continue this discussion/experiments in #4116, I won't close this issue after merging the PR

File: docs/requirements-doc.txt
Comment (dchigarev) [inline docs/requirements-doc.txt:19]: I won't close the #4115 issue, so we can monitor the state of this FIXME there

File: docs/release_notes/release_notes-0.14.0.rst
Comment (dchigarev) [inline docs/release_notes/release_notes-0.14.0.rst:28]: just noticed this. Should we include such minor PRs in the release notes? If so, maybe the note should be more elaborated?

Comment (dchigarev) [reply docs/release_notes/release_notes-0.14.0.rst:28]: Let's wait on fully green CI before changing this, we can verify that the actual in-code and deps changes work well, so not to wait for complete CI rerun again after changing this docs thing.

Comment (prutskov) [reply docs/release_notes/release_notes-0.14.0.rst]: @dchigarev CI is green, this line was rephrased.

**Review by dchigarev — APPROVED** [review]
Thanks for the changes

**Pull request merged**
